### PR TITLE
main: add tcmu_load_config error check and clean up the code

### PR DESCRIPTION
**Pull request by lxbsz**
Signed-off-by: Xiubo Li <lixiubo@cmss.chinamobile.com>

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -559,10 +559,10 @@ static void cmdproc_thread_cleanup(void *arg)
 
 static void *tcmur_cmdproc_thread(void *arg)
 {
-        int ret;
 	struct tcmu_device *dev = arg;
 	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
 	struct pollfd pfd;
+        int ret;
```
Comment (mikechristie) [inline main.c]: What happened here? I think you might have just cut and paste the original code and as a result took the spaces instead of tabs.

Comment (lxbsz) [inline main.c]: Yes, just copied it here.

**Comment by lxbsz** [comment]
Have updated and rebased to the newest master branch.

File: main.c [hunk 2]
```diff
@@ -782,22 +782,27 @@ static struct option long_options[] = {
...
 	tcmu_cfg = tcmu_config_new();
 	if (!tcmu_cfg)
 		exit(1);
-	tcmu_load_config(tcmu_cfg, NULL);
+	ret = tcmu_load_config(tcmu_cfg, NULL);
+	if (ret == -1)
+		goto err_out;
+
+	/* set the log level */
+	tcmu_set_log_level(tcmu_cfg->log_level);
```
Comment (mikechristie) [inline main.c]: Sorry. I think I missed this on the first viewing of the patch. Why did you add this call here?

It looks like tcmu_load_config does tcmu_parse_options -> tcmu_conf_set_options -> tcmu_set_log_level

Comment (lxbsz) [inline main.c]: Sorry, This just added one comment before the dyn reload PR(And this PR has removed this).
To resolve the confliction I introduced it back again.

Will remove it right now.

Thanks,

**Comment by lxbsz** [comment]
Deleted it.

**Comment by pkalever** [comment]
Although I did not understand the need of altering the placements of various variable declarations in this patch the return value check of tcmu_load_config()  make-sense to me.

verall looks good to me :-)

**Comment by mikechristie** [comment]
Thanks for the review pkalever.

**Comment by lxbsz** [comment]
@pkalever 

For altering the placements of ret, i just keeping this the same with the kernel's code style and make the code looks tidier, that's all.

Thanks,
